tools/cluster/cluster.go

```diff
@@ -309,6 +325,71 @@ func (clu *Cluster) start(dataPath string) error {
 	return nil
 }
 
+func (clu *Cluster) KillNode(nodeIndex int) error {
+	if nodeIndex < len(clu.waspCmds) {
+		process := clu.waspCmds[nodeIndex]
+
+		if process != nil {
+			err := process.Process.Kill()
+
+			if err == nil {
+				clu.waspCmds[nodeIndex] = nil
+			}
+
+			return err
+		}
+
+		return nil
+	}
+	return fmt.Errorf("[cluster] Wasp node with index %d not found", nodeIndex)
```

Comment:
It's more idiomatic to return early and avoid indentation levels. Same for the other functions below. Instead of 

```
if condition {
  do stuff
}
return error
```

it's often better:

```
if !condition {
  return error
}
do stuff
```


```suggestion
	if nodeIndex >= len(clu.waspCmds) {
		return fmt.Errorf("[cluster] Wasp node with index %d not found", nodeIndex)
	}

	process := clu.waspCmds[nodeIndex]
	if process == nil {
		return nil
	}

	err := process.Process.Kill()
	if err != nil {
		return err
	}

	clu.waspCmds[nodeIndex] = nil
```